### ISSUE TITLE
pipeline doesn't recover if page is purged after queue.max_bytes is reached

### ISSUE THREAD

**Issue by jsvd** [elastic/logstash#6626]
setup:

```yml
queue.type: persisted
queue.page_capacity: 5mb
queue.max_bytes: 10mb
```

start logstash with a stalling pipeline:

```
bin/logstash -e "input { generator {} } filter {sleep { time => 1 } }" -w 1 -b 1
```

wait for a bit, the start logstash without the sleep filter:

```
bin/logstash -e "input { generator {} } filter { }" -w 1 -b 1
{
      "sequence" => 1,
    "@timestamp" => 2017-02-01T20:43:03.873Z,
      "@version" => "1",
          "host" => "Joaos-MBP-5.lan",
       "message" => "Hello world!"
}
{
      "sequence" => 2,
    "@timestamp" => 2017-02-01T20:43:03.972Z,
      "@version" => "1",
          "host" => "Joaos-MBP-5.lan",
       "message" => "Hello world!"
}
....
{
      "sequence" => 37448,
    "@timestamp" => 2017-02-01T20:43:05.297Z,
      "@version" => "1",
          "host" => "Joaos-MBP-5.lan",
       "message" => "Hello world!"
}
{
      "sequence" => 37449,
    "@timestamp" => 2017-02-01T20:43:05.297Z,
      "@version" => "1",
          "host" => "Joaos-MBP-5.lan",
       "message" => "Hello world!"
}
{
      "sequence" => 0,
    "@timestamp" => 2017-02-01T20:43:44.280Z,
      "@version" => "1",
          "host" => "Joaos-MBP-5.lan",
       "message" => "Hello world!"
}
# event flow stops
```

**Comment by colinsurprenant** [comment]
I am not sure I understand this. Will have to test locally.

**Comment by colinsurprenant** [comment]
found the problem, solved in #6800

**Comment by suyograo** [comment]
Fixed in https://github.com/elastic/logstash/pull/6800
